attachments/views.py

```diff
@@ -42,19 +42,31 @@ def attach(request, session_id):
                     os.makedirs(temp_dir, mode=settings.FILE_UPLOAD_DIRECTORY_PERMISSIONS)
                 else:
                     os.makedirs(temp_dir)
-            fd, path = tempfile.mkstemp(dir=temp_dir)
-            with os.fdopen(fd, 'wb') as fp:
-                for chunk in f.chunks():
+
+            upload, created = Upload.objects.get_or_create(file_name=f.name, file_size=f.size, session=session, completed=False)
+            if created:
+                fd, path = tempfile.mkstemp(dir=temp_dir)
+                upload.file_path = path
+                upload.save()
+                opened_file = os.fdopen(fd, 'wb')
+            else:
+                opened_file = open(upload.file_path, 'ab')
+            with opened_file as fp:
+                for idx, chunk in enumerate(f.chunks()):
+                    if not created and idx < upload.chunk_index_to_resume_on:
+                        continue
                     fp.write(chunk)
+                    upload.chunk_index_to_resume_on = idx + 1
+                    upload.save()
```

Comment:
this seems a touch db heavy lets add a ```if ATTACHMENT_CHUNK_SAVE_POINT and idx %  ATTACHMENT_CHUNK_SAVE_POINT  = 0``` around this

Reply:
lets make ATTACHMENT_CHUNK_SAVE_POINT not sure how big these chunks are how many but maybe a default of 5 or idk 20 what ever seems somewhat reasonable

Reply:
Sounds good, updated!